# Week 2
## Conditions, Loops and Lists
Let's first take a look at a few examples of what we have seen today.
Try to solve the following quick exercises:
### Loops
Fill the gaps to have a loop printing all numbers between 2 and 10 (included)

In [1]:
for i in range(2, 11):
    print(i)

2
3
4
5
6
7
8
9
10


In [3]:
i=2
while i< 11:
    print(i)
    i+=1

2
3
4
5
6
7
8
9
10


Now let's do every multiple of 12 between 24 and 144 (included).

In [4]:
for i in range( 24, 144+12, 12):
    print(i)

24
36
48
60
72
84
96
108
120
132
144


In [5]:
i= 24
while i< 144+12:
    print(i)
    i+=12

24
36
48
60
72
84
96
108
120
132
144


Slightly harder: 
If we list all the natural numbers below 10 that are multiples of 3, we get 3, 6 and 9. The sum of these multiples is 18.
What's the sum of all the multiples of 3 below 1000?

In [7]:
#put your code here
val = sum([x for x in range(0,1000,3)])
print(val)
    

166833


The solution should be 166833.
### Conditions
Now let's do the same, but you have to use the following line:
```if i%3==0:```
which mean "if i is a multiple of 3, then"

In [11]:
#put your code here
val = sum([x for x in range(0,1000,1) if x%3 == 0])
print(val)

166833


If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
What is the sum of all the multiples of 3 AND 5 below 1000?

In [14]:
#put your code here
val = sum([x for x in range(0,1000,1) if x%3 == 0 or x%5 == 0])
print(val)


233168


If you think you have it right, you just solved the easiest enigma from the Euler project. 
You can check your answer here: https://projecteuler.net/problem=1

### Lists
Now we're gonna see a few different way to define lists and other structures:

In [15]:
#You can define it in one line:
[1, 2, 3, 4]

[1, 2, 3, 4]

In [16]:
#if you open a parenthesis/square bracket or other, you can use multiple lines until you close it.
[
    1,
    2,
    3,
    4
]

[1, 2, 3, 4]

In [9]:
#If you know how long your list is gonna be, you can use a for loop inside the brackets:
[i for i in range(1,5)]

[1, 2, 3, 4]

In [17]:
#you can even use different lists inside another one:
my_list = [i for i in range(1,5)]
[i*0.2 for i in my_list]

[0.2, 0.4, 0.6000000000000001, 0.8]

## A few list tricks:
### zip
To go over multiple lists at the same time. Both lists have to have same length.

In [18]:
names = ['Fred', 'Julia', 'Henry', 'Phoebe', 'Martin', 'Lisa']
ages = [12, 34, 59, 27, 21, 31]
for name, age in zip(names, ages):
    print(f"{name} is {age} years old.")

Fred is 12 years old.
Julia is 34 years old.
Henry is 59 years old.
Phoebe is 27 years old.
Martin is 21 years old.
Lisa is 31 years old.


### enumerate
To get both the index and the element as the same time:

In [19]:
for idx, name in enumerate(names):
    print(f"index {idx}: {name}")

index 0: Fred
index 1: Julia
index 2: Henry
index 3: Phoebe
index 4: Martin
index 5: Lisa


In [20]:
#Or both at the same time:
for idx, (name, age) in enumerate(zip(names, ages)):
    print(f"index {idx}: {name} - {age}")

index 0: Fred - 12
index 1: Julia - 34
index 2: Henry - 59
index 3: Phoebe - 27
index 4: Martin - 21
index 5: Lisa - 31


### break and continue
If you want to get out of a loop, you can use the keyword break
If you want to ignore the rest of the loop, you can use the keyword continue

In [21]:
for name, age in zip(names, ages):
    
    print(name)
    
    if age>30: continue
    
    print(f"{age}<=30")
    
    if name=='Martin': break
    
    print('next loop')

Fred
12<=30
next loop
Julia
Henry
Phoebe
27<=30
next loop
Martin
21<=30


# Today's exercise: 
## Basic Authentication system with passwords
You are given a list of logins and the corresponding passwords.
Considering a list of connection attempts, with the form [login, used password, date of connection, ip], find the proportion of successful authentications.

In [22]:
logins = ['maurice_binyomin@email.com', 'rhea_yonah@email.com', 'amora_yehudis@email.com', 'blair_trany@email.com', 'noah_alec@email.com', 'annabelle_taylor@email.com', 'shia_joshua@email.com', 'andy_helen@email.com', 'enrique_aleah@email.com', 'aurora_jael@email.com', 'bryan_keily@email.com', 'arielle_saul@email.com', 'jimena_tess@email.com', 'genevieve_maximilian@email.com', 'alisha_naomi@email.com', 'vincenzo_nikhil@email.com']
passwords = ['MBINYOMIN55', '20041985', 'AYEHUDIS54', 'BTrany17', 'NOAH1970', '21051959', 'Shia1981', '07081980', 'EAleah46', 'Ariyah4Ever', 'JAYLAH4Ever', 'ASAUL82', '17121974', '21091958', '05011953', 'VINCENZO1956']
connections = [['amora_yehudis@email.com', 'AYEHUDIS54', '02:11:12 07-01-2023', '148.079.136.032'],['blair_trany@email.com', '11071961', '01:39:16 08-08-2021', '105.086.125.025'],['aurora_jael@email.com', 'Ariyah4Ever', '17:13:53 09-12-2022', '009.092.032.192'],['bryan_keily@email.com', 'BRYAN2002', '02:11:12 26-05-2021', '148.079.136.032'],['aurora_jael@email.com', 'Ariyah4Ever', '02:11:12 21-09-2023', '148.079.136.032'],['maurice_binyomin@email.com', '08041991', '00:49:28 04-04-2021', '055.015.143.072'],['shia_joshua@email.com', '27071981', '20:40:49 22-10-2020', '125.248.092.065'],['alisha_naomi@email.com', '05011953', '10:29:08 29-09-2023', '066.071.213.237'],['noah_alec@email.com', 'qwerty', '19:55:33 13-04-2023', '045.147.072.009'],['arielle_saul@email.com', '15031951', '15:00:56 01-12-2019', '242.058.196.202'],['andy_helen@email.com', 'ANDY1980', '11:24:40 25-04-2021', '228.067.221.099'],['amora_yehudis@email.com', '05101984', '21:55:40 04-07-2022', '028.177.180.049'],['amora_yehudis@email.com', 'qwerty', '03:13:54 27-04-2022', '028.177.180.049'],['rhea_yonah@email.com', '20041985', '23:38:16 09-01-2021', '185.103.054.227'],['blair_trany@email.com', 'BTrany17', '02:11:12 16-11-2022', '148.079.136.032'],['jimena_tess@email.com', '17121974', '01:29:28 26-07-2022', '129.238.111.240'],['maurice_binyomin@email.com', 'Idy4Ever', '17:25:38 13-07-2020', '055.015.143.072'],['bryan_keily@email.com', 'BRYAN2002', '07:23:45 26-01-2021', '122.155.100.205'],['bryan_keily@email.com', '21082002', '07:40:34 04-02-2020', '122.155.100.205'],['genevieve_maximilian@email.com', 'GENEVIEVE1958', '14:51:58 25-09-2020', '210.039.002.066'],['noah_alec@email.com', 'NALEC99', '03:52:27 12-05-2022', '045.147.072.009'],['noah_alec@email.com', 'qwerty', '02:11:12 16-07-2021', '148.079.136.032'],['jimena_tess@email.com', 'Matthias4Ever', '17:13:48 03-11-2021', '174.184.122.111'],['arielle_saul@email.com', 'ASAUL4', '00:48:30 13-04-2021', '242.058.196.202'],['bryan_keily@email.com', '21082002', '02:32:16 29-06-2020', '122.155.100.205'],['noah_alec@email.com', "Amar'e4Ever", '02:11:12 25-10-2022', '148.079.136.032'],['rhea_yonah@email.com', '20041985', '05:46:43 18-08-2020', '185.103.054.227'],['genevieve_maximilian@email.com', '22042006', '01:44:45 13-04-2023', '210.039.002.066'],['rhea_yonah@email.com', '20041985', '09:02:55 12-07-2023', '185.103.054.227'],['blair_trany@email.com', 'Jenny4Ever', '02:11:12 04-10-2019', '148.079.136.032'],['blair_trany@email.com', 'Blair1961', '22:51:27 28-03-2022', '118.131.096.002'],['rhea_yonah@email.com', '20041985', '20:43:33 14-04-2023', '185.103.054.227'],['bryan_keily@email.com', 'qwerty', '02:11:12 09-02-2023', '148.079.136.032'],['vincenzo_nikhil@email.com', 'VINCENZO1956', '11:47:57 01-02-2022', '039.050.037.059'],['shia_joshua@email.com', 'Shia1981', '09:26:38 14-05-2020', '125.248.092.065'],['annabelle_taylor@email.com', 'GORDON4Ever', '06:20:17 08-08-2020', '202.236.031.020'],['bryan_keily@email.com', 'TASNIM4Ever', '17:03:43 13-12-2020', '122.155.100.205'],['enrique_aleah@email.com', 'EAleah46', '03:54:21 04-10-2022', '136.206.109.018'],['maurice_binyomin@email.com', '26031966', '18:00:16 20-12-2022', '055.015.143.072'],['genevieve_maximilian@email.com', '21091958', '21:27:44 03-12-2019', '210.039.002.066'],['aurora_jael@email.com', 'Aurora1957', '06:34:19 22-08-2023', '009.092.032.192'],['arielle_saul@email.com', 'ASAUL82', '16:55:16 19-02-2023', '242.058.196.202'],['maurice_binyomin@email.com', 'MBINYOMIN55', '00:29:45 14-12-2021', '055.015.143.072'],['genevieve_maximilian@email.com', '21091958', '16:36:23 13-11-2022', '210.039.002.066'],['arielle_saul@email.com', 'Carla4Ever', '13:07:19 26-04-2022', '242.058.196.202'],['blair_trany@email.com', 'BTrany17', '04:44:41 02-03-2019', '118.131.096.002'],['arielle_saul@email.com', 'ASAUL82', '22:09:12 04-03-2020', '242.058.196.202'],['arielle_saul@email.com', 'ASAUL82', '10:28:27 03-08-2023', '242.058.196.202'],['shia_joshua@email.com', 'qwerty', '01:31:57 12-05-2019', '125.248.092.065'],['jimena_tess@email.com', '17121974', '02:11:12 19-05-2019', '148.079.136.032'],['enrique_aleah@email.com', 'EAleah46', '03:19:47 16-05-2020', '136.206.109.018'],['rhea_yonah@email.com', '20041985', '22:41:08 07-11-2021', '185.103.054.227'],['alisha_naomi@email.com', '05011953', '13:07:42 03-06-2021', '066.071.213.237'],['andy_helen@email.com', '07081980', '18:55:09 22-10-2022', '228.067.221.099'],['vincenzo_nikhil@email.com', 'VINCENZO1956', '07:32:05 09-03-2020', '177.044.067.084'],['annabelle_taylor@email.com', 'ATaylor89', '19:18:45 23-08-2021', '202.236.031.020'],['blair_trany@email.com', '11071961', '18:17:40 11-10-2023', '118.131.096.002'],['blair_trany@email.com', 'BTrany17', '13:40:14 08-06-2019', '248.147.169.084'],['maurice_binyomin@email.com', '24081982', '17:49:58 25-06-2019', '055.015.143.072'],['shia_joshua@email.com', 'Samira4Ever', '01:41:35 11-03-2023', '125.248.092.065'],['jimena_tess@email.com', '17121974', '02:11:12 24-08-2020', '148.079.136.032'],['blair_trany@email.com', '24021994', '02:53:23 13-08-2021', '118.131.096.002'],['jimena_tess@email.com', '17121974', '02:11:12 11-09-2023', '148.079.136.032'],['enrique_aleah@email.com', 'EAleah46', '14:14:51 12-02-2023', '136.206.109.018'],['aurora_jael@email.com', 'Aurora1957', '19:49:49 15-08-2022', '009.092.032.192'],['noah_alec@email.com', 'NOAH1970', '21:47:11 20-06-2023', '206.160.131.192'],['bryan_keily@email.com', '02052009', '21:15:08 14-12-2021', '122.155.100.205'],['andy_helen@email.com', '07081980', '08:53:39 18-06-2021', '228.067.221.099'],['vincenzo_nikhil@email.com', 'VINCENZO1956', '02:35:32 02-05-2023', '177.044.067.084'],['rhea_yonah@email.com', '20041985', '21:38:11 10-03-2023', '185.103.054.227'],['rhea_yonah@email.com', '20041985', '08:04:45 25-05-2023', '185.103.054.227'],['jimena_tess@email.com', '17121974', '02:11:12 07-11-2022', '148.079.136.032'],['bryan_keily@email.com', 'BKeily57', '10:17:46 07-07-2023', '122.155.100.205'],['bryan_keily@email.com', 'BKeily99', '02:11:12 08-01-2019', '148.079.136.032'],['shia_joshua@email.com', 'Shia1981', '14:00:10 25-07-2020', '125.248.092.065'],['vincenzo_nikhil@email.com', '03041956', '04:32:00 06-10-2019', '177.044.067.084'],['genevieve_maximilian@email.com', '21091958', '16:32:09 02-08-2021', '210.039.002.066'],['maurice_binyomin@email.com', 'MBINYOMIN55', '11:23:09 07-08-2019', '055.015.143.072'],['annabelle_taylor@email.com', '21051959', '13:52:40 18-04-2022', '202.236.031.020'],['genevieve_maximilian@email.com', '21091958', '05:31:54 29-11-2019', '210.039.002.066'],['bryan_keily@email.com', 'qwerty', '02:11:12 22-04-2020', '148.079.136.032'],['rhea_yonah@email.com', 'qwerty', '01:20:07 25-11-2022', '185.103.054.227'],['noah_alec@email.com', 'qwerty', '20:43:25 09-06-2019', '045.147.072.009'],['annabelle_taylor@email.com', '21051959', '02:11:12 09-12-2022', '148.079.136.032'],['amora_yehudis@email.com', 'Autumn4Ever', '23:04:40 06-06-2022', '028.177.180.049'],['bryan_keily@email.com', '21082002', '23:45:06 03-06-2021', '122.155.100.205'],['noah_alec@email.com', '12011970', '21:42:58 01-10-2021', '205.002.237.112'],['enrique_aleah@email.com', 'ENRIQUE1968', '09:25:46 10-05-2019', '236.178.184.039'],['annabelle_taylor@email.com', '21051959', '12:29:39 01-10-2019', '202.236.031.020'],['maurice_binyomin@email.com', 'ROSELYN4Ever', '23:45:04 14-05-2022', '055.015.143.072'],['rhea_yonah@email.com', 'RYonah7', '17:58:38 10-01-2020', '185.103.054.227'],['bryan_keily@email.com', '20121955', '21:31:00 27-10-2023', '201.131.255.206'],['noah_alec@email.com', 'NOAH1970', '00:00:32 02-06-2022', '045.147.072.009'],['vincenzo_nikhil@email.com', 'VINCENZO1956', '10:38:42 16-01-2020', '177.044.067.084'],['rhea_yonah@email.com', '20041985', '08:30:16 10-08-2021', '185.103.054.227'],['andy_helen@email.com', 'ANDY1980', '03:19:52 11-09-2022', '228.067.221.099'],['rhea_yonah@email.com', '20041985', '02:11:12 19-03-2023', '148.079.136.032'],['noah_alec@email.com', 'NOAH1970', '08:30:49 11-10-2021', '045.147.072.009'],['noah_alec@email.com', 'NOAH1970', '02:11:12 27-12-2019', '148.079.136.032'],['jimena_tess@email.com', '12111959', '11:15:46 22-04-2022', '129.238.111.240'],['blair_trany@email.com', '18051972', '19:23:11 25-04-2023', '118.131.096.002'],['enrique_aleah@email.com', '04091968', '20:28:14 07-12-2020', '136.206.109.018'],['genevieve_maximilian@email.com', '21091958', '23:25:33 08-03-2019', '210.039.002.066'],['arielle_saul@email.com', 'ASAUL22', '00:23:15 09-08-2021', '242.058.196.202'],['arielle_saul@email.com', 'ASAUL81', '08:37:10 17-04-2021', '242.058.196.202'],['arielle_saul@email.com', '25101996', '03:24:40 18-11-2021', '242.058.196.202'],['blair_trany@email.com', 'BTrany17', '16:19:48 24-07-2019', '118.131.096.002'],['rhea_yonah@email.com', 'RYonah19', '20:08:11 26-09-2022', '112.229.044.032'],['aurora_jael@email.com', 'Ariyah4Ever', '22:13:22 19-04-2021', '009.092.032.192'],['alisha_naomi@email.com', 'qwerty', '02:11:12 08-02-2023', '148.079.136.032'],['vincenzo_nikhil@email.com', 'qwerty', '02:36:42 29-07-2023', '177.044.067.084'],['shia_joshua@email.com', 'SJOSHUA78', '13:02:15 06-05-2022', '125.248.092.065'],['amora_yehudis@email.com', '06042005', '02:11:12 18-01-2019', '148.079.136.032'],['blair_trany@email.com', 'BTrany17', '10:59:33 15-09-2023', '118.131.096.002'],['jimena_tess@email.com', '17121974', '15:36:14 28-05-2022', '129.238.111.240'],['bryan_keily@email.com', '13072001', '01:06:45 02-06-2022', '122.155.100.205'],['andy_helen@email.com', 'ANDY1980', '05:07:55 17-01-2021', '228.067.221.099'],['vincenzo_nikhil@email.com', '03041956', '01:39:45 29-03-2023', '177.044.067.084'],['aurora_jael@email.com', 'Aurora1957', '04:00:24 11-05-2022', '009.092.032.192'],['bryan_keily@email.com', 'qwerty', '20:59:44 06-04-2022', '122.155.100.205'],['shia_joshua@email.com', 'SJOSHUA77', '02:11:12 09-09-2022', '148.079.136.032'],['annabelle_taylor@email.com', '21051959', '02:11:12 17-06-2023', '148.079.136.032'],['noah_alec@email.com', '12011970', '01:01:10 09-09-2021', '045.147.072.009'],['enrique_aleah@email.com', 'EAleah65', '02:32:18 23-03-2019', '161.067.239.193'],['amora_yehudis@email.com', 'AYEHUDIS32', '03:27:21 10-10-2019', '028.177.180.049'],['maurice_binyomin@email.com', 'MAURICE2005', '03:24:39 01-04-2020', '055.015.143.072'],['alisha_naomi@email.com', '05011953', '13:38:38 11-10-2023', '066.071.213.237'],['aurora_jael@email.com', 'AMARI4Ever', '17:48:22 13-09-2022', '201.056.045.213'],['enrique_aleah@email.com', 'EAleah46', '11:04:26 12-11-2020', '136.206.109.018'],['noah_alec@email.com', '08031953', '10:23:03 02-07-2020', '045.147.072.009'],['andy_helen@email.com', '07081980', '23:37:18 02-01-2020', '228.067.221.099'],['shia_joshua@email.com', '27071981', '02:11:12 26-10-2020', '148.079.136.032'],['andy_helen@email.com', 'Bianca4Ever', '07:43:23 13-06-2019', '228.067.221.099'],['jimena_tess@email.com', '17121974', '13:05:34 19-04-2023', '129.238.111.240'],['genevieve_maximilian@email.com', 'GMAXIMILIAN75', '19:01:45 15-12-2020', '210.039.002.066'],['bryan_keily@email.com', 'qwerty', '15:20:05 05-02-2019', '122.155.100.205'],['alisha_naomi@email.com', 'ANAOMI1', '22:38:55 04-08-2019', '066.071.213.237'],['aurora_jael@email.com', '15081957', '08:52:10 16-01-2019', '009.092.032.192'],['annabelle_taylor@email.com', '24111975', '02:11:12 17-05-2023', '148.079.136.032'],['maurice_binyomin@email.com', 'MBINYOMIN55', '21:10:44 21-02-2019', '055.015.143.072'],['blair_trany@email.com', 'RYDER4Ever', '05:04:43 27-07-2022', '118.131.096.002'],['aurora_jael@email.com', 'Ariyah4Ever', '22:10:26 01-07-2021', '009.092.032.192'],['noah_alec@email.com', 'NOAH1970', '02:11:12 12-09-2022', '148.079.136.032'],['alisha_naomi@email.com', 'ANAOMI74', '10:40:37 11-10-2020', '069.241.078.203'],['annabelle_taylor@email.com', 'Annabelle1975', '08:53:41 28-08-2022', '202.236.031.020'],['vincenzo_nikhil@email.com', 'VINCENZO1956', '03:41:33 11-11-2023', '177.044.067.084'],['rhea_yonah@email.com', 'Preston4Ever', '14:55:44 08-06-2022', '185.103.054.227'],['genevieve_maximilian@email.com', '21091958', '16:22:00 28-10-2021', '210.039.002.066'],['aurora_jael@email.com', 'Ariyah4Ever', '21:10:12 05-04-2019', '009.092.032.192'],['rhea_yonah@email.com', '20041985', '07:51:23 03-07-2022', '185.103.054.227'],['shia_joshua@email.com', 'Shia1981', '18:36:30 06-12-2023', '125.248.092.065'],['genevieve_maximilian@email.com', '21091958', '23:12:49 03-12-2023', '210.039.002.066'],['maurice_binyomin@email.com', 'MAURICE2005', '02:11:12 07-04-2022', '148.079.136.032'],['alisha_naomi@email.com', '05011953', '09:40:03 08-05-2019', '066.071.213.237'],['jimena_tess@email.com', 'JIMENA1959', '20:10:34 01-05-2023', '129.238.111.240'],['bryan_keily@email.com', 'JAYLAH4Ever', '04:43:56 24-07-2021', '122.155.100.205'],['annabelle_taylor@email.com', 'ATaylor45', '08:57:42 23-03-2023', '202.236.031.020'],['shia_joshua@email.com', '23051967', '02:11:12 29-03-2019', '148.079.136.032'],['noah_alec@email.com', '03081975', '03:40:46 20-03-2019', '045.147.072.009'],['shia_joshua@email.com', 'Shia1981', '00:10:23 27-04-2023', '125.248.092.065'],['maurice_binyomin@email.com', 'MBINYOMIN55', '18:24:30 25-03-2023', '055.015.143.072'],['shia_joshua@email.com', 'Shia1981', '05:43:43 04-10-2019', '125.248.092.065'],['alisha_naomi@email.com', '05011953', '22:28:27 19-04-2020', '066.071.213.237'],['amora_yehudis@email.com', 'LINDA4Ever', '18:32:45 20-12-2020', '028.177.180.049'],['jimena_tess@email.com', '27062005', '13:30:47 21-10-2021', '129.238.111.240'],['rhea_yonah@email.com', 'Rhea1985', '18:48:43 29-02-2019', '185.103.054.227'],['enrique_aleah@email.com', 'qwerty', '15:14:01 10-10-2020', '136.206.109.018'],['bryan_keily@email.com', '01061987', '20:33:34 10-10-2020', '122.155.100.205'],['noah_alec@email.com', 'Kabir4Ever', '00:02:17 18-09-2019', '045.147.072.009'],['noah_alec@email.com', 'NOAH1970', '01:20:13 02-10-2021', '045.147.072.009'],['maurice_binyomin@email.com', 'MAURICE2005', '20:09:10 21-08-2019', '055.015.143.072'],['noah_alec@email.com', 'NOAH1970', '22:14:01 28-07-2021', '045.147.072.009'],['alisha_naomi@email.com', 'HAREEM4Ever', '14:13:42 21-05-2020', '225.135.215.119'],['andy_helen@email.com', '07081980', '18:38:20 02-09-2020', '228.067.221.099'],['andy_helen@email.com', '07081980', '20:09:45 03-03-2020', '228.067.221.099'],['aurora_jael@email.com', 'Aurora1957', '18:57:33 02-04-2023', '009.092.032.192'],['amora_yehudis@email.com', 'AYEHUDIS54', '02:11:12 09-03-2020', '148.079.136.032'],['amora_yehudis@email.com', '20092009', '15:34:07 24-01-2023', '028.177.180.049'],['maurice_binyomin@email.com', 'MBINYOMIN55', '10:46:12 12-03-2019', '174.012.119.010'],['maurice_binyomin@email.com', 'qwerty', '06:21:55 07-10-2020', '055.015.143.072'],['noah_alec@email.com', 'NOAH1970', '02:50:04 23-09-2020', '045.147.072.009'],['andy_helen@email.com', '07081980', '16:26:42 14-10-2020', '228.067.221.099'],['andy_helen@email.com', '07081980', '02:47:19 28-06-2019', '228.067.221.099'],['andy_helen@email.com', '07081980', '05:45:05 29-07-2023', '228.067.221.099'],['shia_joshua@email.com', 'Shia1981', '02:11:12 21-11-2019', '148.079.136.032'],['rhea_yonah@email.com', '20041985', '04:46:12 09-04-2020', '185.103.054.227'],['vincenzo_nikhil@email.com', 'VINCENZO1956', '16:42:58 11-01-2023', '177.044.067.084'],['andy_helen@email.com', 'AHelen74', '02:11:12 28-06-2021', '148.079.136.032'],['arielle_saul@email.com', 'CESAR4Ever', '19:10:01 04-10-2023', '242.058.196.202'],['annabelle_taylor@email.com', 'ATaylor71', '02:01:59 11-02-2023', '202.236.031.020'],['andy_helen@email.com', '07081980', '02:11:12 24-04-2020', '148.079.136.032'],['bryan_keily@email.com', '21082002', '17:33:58 17-10-2021', '211.099.239.154'],['alisha_naomi@email.com', 'ALISHA1953', '06:57:41 28-01-2023', '066.071.213.237'],['amora_yehudis@email.com', 'AYEHUDIS54', '02:11:12 10-08-2021', '148.079.136.032'],['aurora_jael@email.com', 'Ariyah4Ever', '02:41:00 14-09-2021', '009.092.032.192'],['vincenzo_nikhil@email.com', 'VINCENZO1956', '07:58:04 20-11-2019', '177.044.067.084'],['bryan_keily@email.com', 'BKeily29', '23:44:42 25-09-2020', '122.155.100.205'],['blair_trany@email.com', '11071961', '02:11:12 02-07-2023', '148.079.136.032'],['enrique_aleah@email.com', 'EAleah46', '18:57:18 09-11-2022', '136.206.109.018'],['alisha_naomi@email.com', 'KYLEE4Ever', '06:01:14 18-07-2020', '252.138.066.086']]

print(len(logins), len(passwords), len(connections))


16 16 200


In [24]:
print(len(logins), len(passwords))

16 16


In [25]:
print([1,2,3][0:1])

[1]


In [31]:




accounts = [[x,y] for x,y in zip(logins, passwords)]


#print(accounts)

success = 0
tries = len(connections)

# put your code here
for connection in connections:
    if connection[0:2] in accounts:
        #print(connection[0:2])
        success += 1
    else:
        pass
    
percent = success / tries * 100

print(f"The percentage of successful authentications is {percent}%")

The percentage of successful authentications is 46.0%


### Bonus question: There is a Hacker in the connections attempts, what is his IP adress? 
<details>
  <summary>Hint:</summary>
  
  What's the ip adress with the most connections attempts?
  
</details>

In [35]:
# put your code here

hash_map = {}










for connection in connections:
    if connection[3] in hash_map:
        hash_map[connection[3]] += 1
    else :
        hash_map[connection[3]] = 1


print(hash_map)

{'148.079.136.032': 32, '105.086.125.025': 1, '009.092.032.192': 10, '055.015.143.072': 12, '125.248.092.065': 9, '066.071.213.237': 7, '045.147.072.009': 12, '242.058.196.202': 10, '228.067.221.099': 12, '028.177.180.049': 6, '185.103.054.227': 14, '129.238.111.240': 6, '122.155.100.205': 13, '210.039.002.066': 10, '174.184.122.111': 1, '118.131.096.002': 8, '039.050.037.059': 1, '202.236.031.020': 7, '136.206.109.018': 7, '177.044.067.084': 9, '248.147.169.084': 1, '206.160.131.192': 1, '205.002.237.112': 1, '236.178.184.039': 1, '201.131.255.206': 1, '112.229.044.032': 1, '161.067.239.193': 1, '201.056.045.213': 1, '069.241.078.203': 1, '225.135.215.119': 1, '174.012.119.010': 1, '211.099.239.154': 1, '252.138.066.086': 1}


In [ ]:
# convert date time objects to seconds 
